### ISSUE TITLE
Split actix-core

### ISSUE THREAD

**Issue by fafhrd91** [actix/actix#70]
I'd like to split actix into two packages

1. actix-core - which should depend only on futures
2. actix - kind of runtime based on tokio

**Comment by vhdirk** [comment]
I assume this would allow swapping tokios event loop with, for instance, glib?

**Comment by krircc** [comment]
name is important.

1. actix - which should depend only on futures
2. actix-runtime  - kind of runtime based on tokio

is some idea

**Comment by xvny** [comment]
@fafhrd91 I agree with @krircc if actix-core is the core actor systerm for many crates use,  the 'actix' name more sences .  also 'actix' name base on tokio will be influenced maybe some day that 'tokio' be found some problem.

**Comment by moosingin3space** [comment]
I support `actix-core` as a name, because Actix is used, confusingly, to refer to both the actor system and web framework.

**Comment by anp** [comment]
actix -> actix-{runtime,rt}-tokio -> actix-web?

**Comment by luover** [comment]
The new Actor framework [riker](https://github.com/riker-rs/riker) not use the tokio.

**Comment by fafhrd91** [comment]
riker looks interesting, but I am not sure if rust needs actors system at all.

**Comment by fafhrd91** [comment]
sorry, I don't have time atm. if anyone wants to work on this, I'll give access to repo. also @DoumanAsh can help :)

**Comment by botika** [comment]
I can work on this. But I will need more details.

**Comment by nitingupta910** [comment]
> riker looks interesting, but I am not sure if rust needs actors system at all.

I'm confused. Isn't actix an actor framework itself?
